README.md

```diff
@@ -237,3 +237,27 @@ If you want to grant members of other groups ability to view the results, you ca
 names in the django settings using the `XBLOCK_POLL_EXTRA_VIEW_GROUPS` setting, for example:
 
     XBLOCK_POLL_EXTRA_VIEW_GROUPS = ['poll_staff']
+
+## XBlock Translations
+
+### Add translations to other languages.
+
+Create a set of language directories for each of your locales within the "translations" directory. You may specify either a general language code or a specific language locale code for the name of each directory.
+Include an "LC_MESSAGES" directory with each language directory.
+
+../my_xblock/translations/ar/LC_MESSAGES/
+../my_xblock/translations/es-es/LC_MESSAGES/
+Create a domain file named "text.po". You can use the [i18n-tools](https://github.com/edx/i18n-tools) for extraction and compilation process.
+
+### View The XBlock In Different Language
+Before installing the xblock, you need to Compile the .po file into .mo. you can use the [i18n_tool](https://github.com/edx/i18n-tools) or the command given below:
+   msgfmt text.po -o text.mo
+After successful installation, You can view the translated xblock via adding query param '/?preview-lang=<lang_code>' in url.
+Note: you can clear the language by adding the query param "/?clear-lang" in url.
+
+### Update the Existing Translation Files
+In order to update the existing translations, You need to extract the strings from source files into .PO and compile them as described above.
+for extraction, you can use the [i18n_tool](https://github.com/edx/i18n-tools)
+
+You can find detailed/additional information here:
+[Xblock-translations](http://edx.readthedocs.io/projects/xblock-tutorial/en/latest/edx_platform/edx_lms.html#adding-translated-strings-to-your-xblock)
```

Comment:
@asadiqbal08 Thanks for adding more detail to the README. It still needs a little bit of polish -- see below for an updated version that you could use (please replace `# FIXME: ...` comments with appropriate content). You can click to edit this comment to get the corresponding Markdown source. If you don't have permission to do that, you can also get it from the raw version of [this gist](https://gist.githubusercontent.com/itsjeyd/7bfff26b0a0f15d9a1ce6fd0a02335ba/raw/41a7ece66395eef35fdad28db069986c76447b86/TRANSLATIONS.md).

## Working with Translations

The following sections provide basic information about common use cases for working with translations. You can find additional information in [this section](http://edx.readthedocs.io/projects/xblock-tutorial/en/latest/edx_platform/edx_lms.html#adding-translated-strings-to-your-xblock) of the [Open edX XBlock Tutorial](http://edx.readthedocs.io/projects/xblock-tutorial/en/latest/index.html).

### Viewing the XBlock in a Different Language

To view this XBlock in a different language you first need to compile the `text.po` file for the language to a file called `text.mo`.  Translation files for a given language are located in the `poll/translations/<locale>/LC_MESSAGES` directory. From the top-level directory of this repository, use the following commands to access the directory and compile the translation file (replacing `<lang_code>` with the appropriate language code):

```sh
cd poll/translations/<lang_code>/LC_MESSAGES/
msgfmt text.po -o text.mo
```

You can then navigate to a poll or survey in the LMS and view a translated version of it by adding `/?preview-lang=<lang_code>` to the URL. Note that user-editable content (such as questions and answers belonging to a poll) is not translated by default. To revert to the default language, add `/?clear-lang` to the URL.

### Adding Translations for Other Languages

To add translations for other languages, create a set of language directories for each of your locales within the `poll/translations` directory. You may specify either a general language code or a specific language locale code for the name of each directory. Include an `LC_MESSAGES` directory with each language directory. For example:

`../poll/translations/ar/LC_MESSAGES/`
`../poll/translations/es-es/LC_MESSAGES/`

For each language, create a domain file named `text.po` and compile it: You can use edX's [i18n-tools](https://github.com/edx/i18n-tools) to extract translations for multiple languages as follows:

```sh
# FIXME: Replace with command(s) for extracting translations
```

Open each `text.po` domain file and, for each `msgid` string, add a corresponding `msgstr` translation. You can then use edX's [i18n-tools](https://github.com/edx/i18n-tools) to compile the translation files as follows:

```sh
# FIXME: Replace with command(s) for compiling translations
```

### Updating Existing Translation Files

After changing existing strings in the source code (or adding new ones), you'll need to update the translation files. To do this, extract and compile the translations as described in the previous section. (There is no need to create additional directories if you are not adding translations for additional languages.)